subsys/bluetooth/host/hci_core.c

```diff
@@ -7185,6 +7185,153 @@ bool bt_addr_le_is_bonded(uint8_t id, const bt_addr_le_t *addr)
 	}
 }
 
+int bt_le_per_adv_set_param(struct bt_le_ext_adv *adv,
+			    const struct bt_le_per_adv_param *param)
+{
+	struct bt_hci_cp_le_set_per_adv_param *cp;
+	struct net_buf *buf;
+	int err;
+
+	if (!atomic_test_bit(adv->flags, BT_ADV_PARAMS_SET)) {
```

Comment:
Remove this check

Reply:
Hmm, can I ask why? In case that this flag isn't set, then we don't have an advertising set allocated, and the command sent in this function will fail. We could rely on the controller error, but why not stop it here?

Reply:
You have an advertising set because you have the advertising set object.

`bt_le_ext_adv_create` returns the object only on success, so the advertising parameters has been set.

`BT_ADV_PARAMS_SET` is only there to handle the random address command not being issued too early inside `bt_le_ext_adv_create` itself.

Reply:
Ah, yes of course :)